src/planscape/planning/views.py

```diff
@@ -0,0 +1,203 @@
+    result = data['properties']
+    result['id'] = data['id']
+    if 'geometry' in data and add_geometry:
+        result['geometry'] = data['geometry']
+    if 'region_name' in result:
+        result['region_name'] = region_to_display_name(result['region_name'])
+    return result
+
+
+#### PLAN(NING AREA) Handlers ####
+
+# Requires a logged in user (from the session), a plan name, a region name, and a geometry.
+def create_plan(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        user = _get_user(request)
+        if user is None:
+            raise ValueError("User must be logged in.")
+        
+        # Get the name of the plan.
+        body = json.loads(request.body)
+        name = body.get('name')
+        if name is None:
+            raise ValueError("Must specify a plan name.")
+
+        # Get the region name; it should be in the human-readable display name format.
+        region_name_input = body.get('region_name')
+        if region_name_input is None:
+            raise ValueError("Region name must be specified.")
+        
+        region_name = display_name_to_region(region_name_input)
+        if region_name is None:
+            raise ValueError("Unknown region_name: " + region_name_input)
+
+        # Get the geometry of the plan.
+        geometry = body.get('geometry')
+        if geometry is None:
+            raise ValueError("Must specify the plan geometry.")
+
+        # Convert to a MultiPolygon if it is a simple Polygon, since the model column type is
+        # MultiPolygon.
+        geometry = _convert_polygon_to_multipolygon(geometry)
+
+        # Create the plan
+        plan = PlanningArea.objects.create(
+            user=user, name=name, region_name=region_name, geometry=geometry)
+        plan.save()
+        return HttpResponse(str(plan.pk))
+    except Exception as e:
+        return HttpResponseBadRequest("Error in create: " + str(e))
+
+
+# Supports deleting a whole list of plans.
+# Plans are deleteable only by their owners.  A list that only partially
+# matches the logged in user will result in no plans being deleted.
+def delete_plan(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        user = _get_user(request)
+        if user is None:
+            raise ValueError("User must be logged in.")
+        user_id = user.pk
+
+        # Get the plans                                                                                                                                 
+        body = json.loads(request.body)
+        plan_id = body.get('id', None)
+        plan_ids = []
+        if plan_id is None:
+            raise ValueError("Must specify plan id")
+        elif isinstance(plan_id, int):
+            plan_ids = [plan_id]
+        elif isinstance(plan_id, list):
+            plan_ids = plan_id
+        else:
+            raise ValueError("Bad plan id: " + plan_id)
```

Comment:
Maybe add the plan_id type in the value error (since it'll be hard to tell why the plan id is bad if it's being parsed as a string instead of an int or something)

Reply:
Updated the text a bit.